src/Reflection/InitializerExprTypeResolver.php

```diff
@@ -924,6 +924,32 @@ public function getPlusType(Expr $left, Expr $right, callable $getTypeCallback):
 			]);
 		}
 
+		if (
+			($arrayType->isSuperTypeOf($leftType)->yes() && $arrayType->isSuperTypeOf($rightType)->no())
+			|| ($arrayType->isSuperTypeOf($rightType)->yes() && $arrayType->isSuperTypeOf($leftType)->no())
+		) {
+			return new ErrorType();
+		}
+
+		if (
+			($arrayType->isSuperTypeOf($leftType)->yes() && $arrayType->isSuperTypeOf($rightType)->maybe())
+			|| ($arrayType->isSuperTypeOf($rightType)->yes() && $arrayType->isSuperTypeOf($leftType)->maybe())
+			|| ($arrayType->isSuperTypeOf($leftType)->maybe() && $arrayType->isSuperTypeOf($rightType)->maybe())
+			|| ($arrayType->isSuperTypeOf($rightType)->maybe() && $arrayType->isSuperTypeOf($leftType)->maybe())
```

Comment:
I feel this can be simplified somehow, but I am not sure yet how.

Reply:
The least you can do here is to extract `$arrayType->isSuperTypeOf($rightType)` and `$arrayType->isSuperTypeOf($leftType)` to a variable. Maybe then it will be apparent how to make it simpler.